share/setup.nsi.in

```diff
@@ -159,30 +159,41 @@ SectionEnd
 # Installer functions
 Function .onInit
     InitPluginsDir
+    SetShellVarContext all
 !if "@WINDOWS_BITS@" == "64"
     ${If} ${RunningX64}
-      ; disable registry redirection (enable access to 64-bit portion of registry)
-      SetRegView 64
-      ReadRegStr $INSTDIR HKCU "${REGKEY}" "Path"
-      StrCmp $INSTDIR "" 0 rununin
-      StrCpy $INSTDIR $PROGRAMFILES64\GridcoinResearch
+      SetRegView 64 ; disable registry redirection (enable access to 64-bit portion of registry)
     ${Else}
-      MessageBox MB_OK|MB_ICONSTOP "Cannot install 64-bit version on a 32-bit system."
+      MessageBox MB_OK|MB_ICONSTOP "Cannot install 64-bit version on a 32-bit system." /SD IDOK
       Abort
     ${EndIf}
-!else
-    ReadRegStr $INSTDIR HKCU "${REGKEY}" "Path"
-    StrCmp $INSTDIR "" 0 rununin
-    StrCpy $INSTDIR $PROGRAMFILES\GridcoinResearch
 !endif
-    rununin:
-      Exec $INSTDIR\uninst.exe
-      Delete $INSTDIR\*"
+    ; Check for HKCU Path registry and prompt for uninstallation if found
+    ReadRegStr $0 HKCU "${REGKEY}" "Path"
+    ${IfThen} $0 == "" ${|} Return ${|}
+    ${IfNot} ${Cmd} \
+        'MessageBox MB_YESNO|MB_ICONEXCLAMATION \
+          "An old version is found to be installed on your system. \
+          Uninstallation will proceed first to migrate registry entries and shortcuts. Continue?" \
```

Comment:
Can we be a little more terse? Maybe

An old version is found on your system and will be uninstalled first to migrate registry entries and shortcuts.  Your wallet data will not be affected. Continue?

Reply:
Done.